manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java

```diff
@@ -70,8 +85,16 @@ public Response getBridges() {
 
     @PUT
     public Response updateBridge(BridgeDTO dto) {
+        failIfNotAuthorized(identity);
         LOGGER.info("Shard wants to update the Bridge with id '{}' with the status '{}'", dto.getId(), dto.getStatus());
         bridgesService.updateBridge(dto);
         return Response.ok().build();
     }
+
+    private void failIfNotAuthorized(SecurityIdentity identity) {
```

Comment:
@r00ta On the next iteration of this work, perhaps we can consider a `Filter` and a CDI annotation to bind the `Filter` to this resource: https://quarkus.io/specs/jaxrs/2.1/index.html#Name_Binding

That would mean that we could annotate this Resource (or other resources) that need a valid Shard identity with something like `@ValidShard` and then we simply encapsulate the logic of checking the shard is valid in the `Filter` we implement. That would align our implementation with other conventions e.g `@Authenticated`

We don't need to do that in this PR, but might be a nice approach if we need to extend the ways in which we check a Shard is valid.

WDYT?

Reply:
https://issues.redhat.com/browse/MGDOBR-245